File: catalog/backends/python-flask/patterns/kafka_producer_route.py
import os
import uuid
import json
import time
from confluent_kafka import Producer
from flask import jsonify

# Pattern: DSM — Kafka producer for pipeline monitoring
# Adapt: replace topic name and payload with domain events
_producer = Producer({"bootstrap.servers": os.environ.get("KAFKA_BOOTSTRAP_SERVERS", "kafka:9092")})


def publish_event():
    event_id = str(uuid.uuid4())
    payload = json.dumps({"id": event_id, "timestamp": time.time()})
    _producer.produce("demo-events", key=event_id, value=payload)
    _producer.flush()
    return jsonify({"status": "sent", "id": event_id})

File: catalog/backends/node-express/patterns/kafka_producer_route.js
const { Kafka } = require('kafkajs');

// Pattern: DSM — Kafka producer for pipeline monitoring
// Adapt: replace topic name and payload with domain events
const kafka = new Kafka({
  brokers: [process.env.KAFKA_BOOTSTRAP_SERVERS || 'kafka:9092'],
});
const producer = kafka.producer();
let connected = false;

async function publishEvent(req, res) {
  if (!connected) { await producer.connect(); connected = true; }
  const id = require('crypto').randomUUID();
  await producer.send({
    topic: 'demo-events',
    messages: [{ key: id, value: JSON.stringify({ id, timestamp: Date.now() }) }],
  });
  res.json({ status: 'sent', id });
}

module.exports = { publishEvent };
